Widen peer dependency range to include React 17

**Summary of proposed feature**

[React 17](https://reactjs.org/blog/2020/10/20/react-v17.html) was released recently with minimal breaking changes. `next-auth` currently supports only `^16.13.1`:

https://github.com/nextauthjs/next-auth/blob/5126f4e3428f2b4b8ebdd38b10066edfdd4ea2fa/package.json#L57-L60

Could this be widened to `^16.13.1 || ^17.0.0`?

**Purpose of proposed feature**

Allows apps using React 17 to also use `next-auth` without peer dependency warnings.

**Detail about proposed feature**
n/a

**Potential problems**
There are [subtle differences](https://reactjs.org/blog/2020/08/10/react-v17-rc.html#changes-to-event-delegation) between React 16 and React 17, so it's worth reviewing to see if any of them might affect the library's functionality.

**Describe any alternatives you've considered**
You could alternatively release a new major version of `next-auth` that peer-depends only on React `^17.0.0`, but since the changes are minimal, you may not want to deprecate support for React 16 going forward.

**Additional context**
n/a

*Please indicate if you are willing and able to help implement the proposed feature.*
Yes.

next-auth doesnt seem to install out of the box on a clean node15/npm7/react17 setup and I wonder if its related to this issue
What am I missing and should this be added to the documentation ?

npm i next-auth
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: myproject@0.1.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.13.1" from next-auth@3.1.0
npm ERR! node_modules/next-auth
npm ERR!   next-auth@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

--- a/package.json
+++ b/package.json
@@ -55,8 +55,8 @@
     "typeorm": "^0.2.24"
   },
   "peerDependencies": {
-    "react": "^16.13.1",
-    "react-dom": "^16.13.1"
+    "react": "^16.13.1 || ^17",
+    "react-dom": "^16.13.1 || ^17"
   },
   "peerOptionalDependencies": {
     "mongodb": "^3.5.9",

